### PR TITLE
Improve ParentClassMethodTypeOverrideGuard performance

### PR DESCRIPTION
**Pull request by staabm**

### CODE REVIEW

File: packages/VendorLocker/ParentClassMethodTypeOverrideGuard.php [hunk 1]
```diff
@@ -37,14 +37,20 @@ public function getParentClassMethod(ClassMethod $classMethod): ?MethodReflectio
 
         /** @var string $methodName */
         $methodName = $this->nodeNameResolver->getName($classMethod);
-        $parentClassReflections = array_merge($classReflection->getParents(), $classReflection->getInterfaces());
```
Comment (staabm) [inline packages/VendorLocker/ParentClassMethodTypeOverrideGuard.php:40]: before this change the whole type hierarchy had to be reflected which can be costly.

with this PR we only reflect what on level at a time

File: packages/VendorLocker/ParentClassMethodTypeOverrideGuard.php [hunk 2]
```diff
@@ -37,14 +37,20 @@ public function getParentClassMethod(ClassMethod $classMethod): ?MethodReflectio
...
+        $parentClassReflection = $classReflection->getParentClass();
+        while  ($parentClassReflection !== null) {
```
Comment (TomasVotruba) [inline packages/VendorLocker/ParentClassMethodTypeOverrideGuard.php]: This should be instanceof check. I'll make a note to make Rector rule to cover these :)

Comment (staabm) [reply packages/VendorLocker/ParentClassMethodTypeOverrideGuard.php]: fixed, thanks

**Review by samsonasik — APPROVED** [review]
Looks good, thank you @staabm

**Pull request merged**